sql/src/main/java/io/crate/expression/symbol/SelectSymbol.java

```diff
@@ -73,6 +77,20 @@ public boolean canBeCasted() {
         return true;
     }
 
+    @Override
+    public Symbol cast(DataType newDataType, boolean tryCast) {
+        if (DataTypes.isCollectionType(newDataType)) {
+            List<Symbol> castSymbols = new ArrayList<>(relation.outputs().size());
+            DataType innerType = ((CollectionType) newDataType).innerType();
+            for (Symbol symbol : relation.outputs()) {
+                castSymbols.add(symbol.cast(innerType, tryCast));
+            }
+            relation.querySpec().outputs(castSymbols);
```

Comment:
This is problematic because the types of the fields won't match the outputs. Why is this necessary now?

Reply:
The idea here was to ad-hoc replace the literals inside the SelectSymbol. Maybe not a good idea, we should go through the regular normalization process instead.

Reply:
Reworked this in https://github.com/crate/crate/pull/7195/commits/91789cf20b1985e31ec7ca2e88ab34dc6716983f